src/rosdiscover/observer/ros1.py

```diff
@@ -0,0 +1,304 @@
+# -*- coding: utf-8 -*-
+import re
+from typing import Dict, Optional
+
+from loguru import logger
+
+from .observer import Observer
+from ..core import Topic
+from ..interpreter import NodeContext, ParameterServer
+
+_GOAL = re.compile(r'(.*)/goal')
+_GOAL_FMT = re.compile(r'(.*)Goal')
+_CANCEL = re.compile(r'(.*)/cancel')
+_STATUS = re.compile(r'(.*)/status')
+_FEEDBACK = re.compile(r'(.*)/feedback')
+_FEEDBACK_FMT = re.compile(r'(.*)Feedback')
+_RESULT = re.compile(r'(.*)/result')
+_RESULT_FMT = re.compile(r'(.*)Result')
+
+_NODES_TO_FILTER_OUT = {'/rosout'}
+_TOPICS_TO_FILTER_OUT = {'/rosout', '/rosout_agg'}
+_SERVICES_TO_FILTER_OUT = {'set_logger_level', 'get_loggers'}
+
+
+class ActionCandidate:
+    goal: Optional[Topic]
```

Comment:
`attrs` should be used to simplify this class.

Reply:
Better yet, the design here could be improved by splitting the different concerns of this class. One class can be a dumb, immutable data structure that represents some given `Action` and the other, `ActionBuilder`, can be used to gather the information necessary to build `Action` objects. If the builder lacks sufficient information, it can raise an exception.

Reply:
Did this through attrs.